sequencer/src/api.rs

```diff
@@ -285,8 +378,9 @@ pub type HandleFromMetrics<N> = Box<
 >;
 
 struct AppState<N: network::Type> {
-    pub submit_state: HotShotSequencingConsensusApi<SeqTypes, Node<N>>,
-    pub query_state: QueryData<SeqTypes, Node<N>, ()>,
+    submit_state: HotShotSequencingConsensusApi<SeqTypes, Node<N>>,
+    query_state: QueryData<SeqTypes, Node<N>, ()>,
+    blocks_by_time: BTreeMap<u64, Vec<u64>>,
```

Comment:
Not sure if worth the effort but internally we could use a dedicated type for the timestamps.

Reply:
Yeah I think this would be good, at least a type alias, but we can do it later.